integration-tests/test/test_slash.py

```diff
@@ -35,39 +40,83 @@ def hex_block_hash(block_hash: bytes) -> bytes:
     return base64.b16encode(block_hash).lower()
 
 
-@pytest.mark.skipif(sys.platform in ('win32', 'cygwin', 'darwin'), reason="Only Linux docker support connection between host and container which node client needs")
-def test_simple_slash(command_line_options: CommandLineOptions, random_generator: Random, docker_client: DockerClient) -> None:
+@contextmanager
+def three_nodes_network_with_node_client(command_line_options: CommandLineOptions, random_generator: Random, docker_client: DockerClient) -> Generator[Tuple[TestingContext, Node, Node, Node, NodeClient], None, None]:
     peers_keypairs = [BONDED_VALIDATOR_KEY_1, BONDED_VALIDATOR_KEY_2]
     with conftest.testing_context(command_line_options, random_generator, docker_client, bootstrap_key=BOOTSTRAP_NODE_KEYS, peers_keys=peers_keypairs) as context, \
         docker_network_with_started_bootstrap(context=context) as bootstrap_node, \
         bootstrap_connected_peer(context=context, bootstrap=bootstrap_node, name='bonded-validator-1', private_key=BONDED_VALIDATOR_KEY_1) as validator1, \
         bootstrap_connected_peer(context=context, bootstrap=bootstrap_node, name='bonded-validator-2', private_key=BONDED_VALIDATOR_KEY_2) as validator2, \
         node_protocol_client(bootstrap_node.network, docker_client, context) as client:
-            contract = '/opt/docker/examples/tut-hello.rho'
+        yield context, bootstrap_node, validator1, validator2, client
+
 
-            validator1.deploy(contract, BONDED_VALIDATOR_KEY_1)
-            blockhash = validator1.propose()
+@pytest.mark.skipif(sys.platform in ('win32', 'cygwin', 'darwin'), reason="Only Linux docker support connection between host and container which node client needs")
```

Comment:
Could you elaborate on what's the exact problem on macOS?  Perhaps I can suggest some solution.

Reply:
The https://github.com/rchain/rchain/blob/dev/integration-tests/test/node_client.py#L138 is a client which needs to be connected by and connect to the rnode. The rnode run in a docker container. The pytest process which run in the host machine in MacOs can not connect to the rnode in the container directly. https://github.com/docker/for-mac/issues/2670

Reply:
As a comment on the linked issue says, just set up port forwarding and (e.g. port 1234) and connect to `localhost:1234`.

Reply:
Then we have to forward every rnode port. Let's do that in another ticket.
https://rchain.atlassian.net/browse/RCHAIN-3772

Reply:
@adaszko Oh. Port forward can only solve the problem with the host connecting to the container but it can not solve the container connect to the host.

Reply:
@adaszko Can we merge this pr first?

Reply:
> Oh. Port forward can only solve the problem with the host connecting to the container but it can not solve the container connect to the host.

And where exactly is that needed?

> Can we merge this pr first?

Just post here the ticket number and fire away.

Reply:
I post the ticket above.https://rchain.atlassian.net/browse/RCHAIN-3772

The core protocol of rnode needs the two way connections between rnodes.

Reply:
> The core protocol of rnode needs the two way connections between rnodes.

Two way connection doesn't mean the network connection needs be **establishable** both ways.  If the connection is established **from pytest towards rnode**, assuming port forwarding is in place, the connection is going to be perfectly capable of transferring data in the reverse direction: **from rnode to pytest**.

Reply:
How about someone interested in running this on MacOS makes the change?

Reply:
@adaszko That's not what I can do about it. It is the design of the rnode. Maybe I am wrong.

Anyway, do you think there is any other places want to change? If not, let's merge this pr first.

Reply:
> Anyway, do you think there is any other places want to change? If not, let's merge this pr first.

Ah, sorry, I said "fire away" but didn't accept the PR ;)